options-user.php

```diff
@@ -112,41 +76,46 @@ function shibboleth_show_user_profile() {
 <?php
 	}
 }
+add_action( 'show_user_profile', 'shibboleth_change_password_profile_link' );
 
 
 /**
- * Ensure profile data isn't updated by the user.  This only applies to accounts that were
- * provisioned through Shibboleth, and only for those user fields marked as 'managed'.
+ * Ensure profile data isn't updated when managed. 
  *
- * @since 1.3
+ * @since 2.3
+ * @param int $user_id
  */
-function shibboleth_personal_options_update() {
-	$user = wp_get_current_user();
+function shibboleth_prevent_managed_fields_update( $user_id ) {
+
+	if ( get_user_meta( $user_id, 'shibboleth_account' ) ) {
+
+		$user = get_user_by('id', $user_id );
 
-	if ( get_user_meta( $user->ID, 'shibboleth_account' ) ) {
 		$managed = shibboleth_get_managed_user_fields();
 
 		if ( in_array( 'first_name', $managed ) ) {
-			add_filter( 'pre_user_first_name', create_function( '$n', 'return $GLOBALS["current_user"]->first_name;' ) );
+			$_POST['first_name']=$user->first_name;
 		}
 
 		if ( in_array( 'last_name', $managed ) ) {
-			add_filter( 'pre_user_last_name', create_function( '$n', 'return $GLOBALS["current_user"]->last_name;' ) );
+			$_POST['last_name']=$user->last_name;
 		}
 
 		if ( in_array( 'nickname', $managed ) ) {
-			add_filter( 'pre_user_nickname', create_function( '$n', 'return $GLOBALS["current_user"]->nickname;' ) );
+			$_POST['nickname']=$user->nickname;
 		}
 
 		if ( in_array( 'display_name', $managed ) ) {
-			add_filter( 'pre_user_display_name', create_function( '$n', 'return $GLOBALS["current_user"]->display_name;' ) );
+			$_POST['display_name']=$user->display_name;
 		}
 
 		if ( in_array( 'email', $managed ) ) {
-			add_filter( 'pre_user_email', create_function( '$e', 'return $GLOBALS["current_user"]->user_email;' ) );
+			$_POST['email']=$user->user_email;
```

Comment:
All 5 of these need some spaces.
```suggestion
			$_POST['email'] = $user->user_email;
```

Reply:
Done in 852a02f.